tests/rados/test_norecovery_flag_functionality.py

```diff
@@ -0,0 +1,299 @@
+        pg_num = pool_obj.get_pg_autoscaler_value(pool_name, "pg_num_target")
+        if status:
+            log.error(
+                f"Scenario4:The pg number is not decreased to after unsetting the norecover flag."
+                f"The pg number is -{pg_num}"
+            )
+            return 1
+        log.info(
+            f"Scenario4:The pg number is decreased after unsetting the norecover flag."
+            f"The pg number is -{pg_num}"
+        )
+        delete_and_healthCheck(rados_obj, pool_name)
+
+        log.info(
+            "Scenario5:Checking the recovery of cluster after setting norecover flag"
+        )
+        utils.configure_osd_flag(ceph_cluster, "set", "norecover")
+        log.info("Scenario5:The norecover is set on the cluster")
+        if not rados_obj.create_pool(pool_name=pool_name, pg_num=16):
+            log.error("Failed to create the  Pool")
+            return 1
+        log.info(f"Scenario5:The pool-{pool_name} is created")
+        pg_num = pool_obj.get_pg_autoscaler_value(pool_name, "pg_num_target")
+        log.info(
+            f"Scenario5:The pg_num before creating data in the {pool_name} is - {pg_num}"
+        )
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=3600)
+        while end_time > datetime.datetime.now():
+            rados_obj.bench_write(pool_name=pool_name)
+            pool_stat = rados_obj.get_cephdf_stats(pool_name=pool_name)
+            out_put = rados_obj.run_ceph_command(cmd="ceph -s")
+            for state in out_put["pgmap"]["pgs_by_state"]:
+                if "recovering" in state["state_name"]:
+                    return 1
+            memory_used = pool_stat["stats"]["percent_used"]
+            log.info(
+                f"The pool-{pool_name} occupied the {memory_used} percentage of memory"
+            )
+            if memory_used >= 14:
+                break
+            time.sleep(10)
+        new_pg_num = pool_obj.get_pg_autoscaler_value(pool_name, "pg_num_target")
+        log.info(
+            f"Scenario5:The pg_num after creating data in the {pool_name} is - {pg_num}"
+        )
+        if pg_num != new_pg_num:
+            log.error(
+                f"Scenario5:During the tests the pg_num changed from {pg_num} to the {new_pg_num}"
+            )
+            return 1
+        log.info(
+            "Scenario5:Recovery not stated on the cluster after setting the recover flag"
+        )
+        delete_and_healthCheck(rados_obj, pool_name)
+    except Exception as e:
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Execution of finally block")
+        if config.get("delete_pool"):
+            method_should_succeed(rados_obj.delete_pool, pool_name)
+            log.info(f"deleted the {pool_name} pool successfully")
+        utils.configure_osd_flag(ceph_cluster, "unset", "norecover")
+        log.info("The norecover is unset on the cluster")
```

Comment:
Please log cluster health in finally.
        rados_obj.log_cluster_health()

Reply:
Added the code.